exec/java-exec/src/main/java/org/apache/drill/exec/physical/PhysicalPlan.java

```diff
@@ -94,6 +94,14 @@ public double totalCost() {
     for (final PhysicalOperator ops : getSortedOperators()) {
       totalCost += ops.getCost();
     }
+
+    // As it turns out, sometimes the total cost can be out of range.
+    // This throws off other code, so clamp the cost at the maximum
+    // double value.
+
+    if (Double.isNaN(totalCost)) {
+      totalCost = Double.MAX_VALUE;
```

Comment:
May be better to replace with 0 or negative number, like -1 (like the second option better). Won't be max double value confusing in this case? It's just nan is not a number and replace it with double max value.